pkg/channel/message_receiver.go

```diff
@@ -136,6 +136,13 @@ func (r *MessageReceiver) Start(ctx context.Context) error {
 }
 
 func (r *MessageReceiver) ServeHTTP(response nethttp.ResponseWriter, request *nethttp.Request) {
+	response.Header().Set("Allow", "POST")
```

Comment:
```
	Header().Set("Allow", "POST, OPTIONS")
``` 

? or why is that different here

Reply:
Because I'm an idiot.

Thanks!

(And, if I'd gotten all my e2e testing working, I'd have noticed this.

Reply:
It actually turns out that my tests are all wrong, because `receiverFunc` is not called at all for OPTIONS requests.

Fixed.